AMZeroDowntimeRollingDeployment e2e test failed on MacBook with M1 chip

<!--

Feel free to ask questions in #prometheus-operator on Kubernetes Slack!

Note: This repository is about prometheus-operator itself, if you have questions about:
- helm installation, go to https://github.com/prometheus-community/helm-charts repository
- kube-prometheus setup, go to https://github.com/prometheus-operator/kube-prometheus

-->

**What happened?**

AMZeroDowntimeRollingDeployment e2e test failed on MacBook with M1 chip

**Did you expect to see something different?**

Yes. Expect AMZeroDowntimeRollingDeployment e2e test success on MacBook with M1 chip.

**How to reproduce it (as minimally and precisely as possible)**:

By using Prometheus Operator latest main branch, run `make test-e2e` on MacBook with M1 chip. 

**Environment**

* Docker information:

```
Client:
 Cloud integration: v1.0.28
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:04:45 2022
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.11.0 (83626)
 Engine:
  Version:          20.10.17
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       a89b842
  Built:            Mon Jun  6 23:01:01 2022
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.6
  GitCommit:        10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc:
  Version:          1.1.2
  GitCommit:        v1.1.2-0-ga916309
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.3", GitCommit:"aef86a93758dc3cb2c658dd9657ab4ad4afc21cb", GitTreeState:"clean", BuildDate:"2022-07-13T14:30:46Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.0", GitCommit:"4ce5a8954017644c5420bae81d72b09b735c21f0", GitTreeState:"clean", BuildDate:"2022-05-19T15:42:59Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/arm64"}
```

* KinD version:

`kind v0.14.0 go1.18.2 darwin/arm64`

* Kubernetes cluster kind:

`kind create cluster`

* `make test-e2e` logs:

```
=== RUN   TestAllNS/x/AMZeroDowntimeRollingDeployment
    alertmanager_test.go:566: waiting for service alertmanager-webhook to become ready timed out: timed out waiting for the condition
--- FAIL: TestAllNS (334.21s)
    --- FAIL: TestAllNS/x (300.08s)
        --- FAIL: TestAllNS/x/AMZeroDowntimeRollingDeployment (300.08s)
```

* `alertmanager-webhook` pod image:

`quay.io/coreos/prometheus-alertmanager-test-webhook:latest`

* `alertmanager-webhook` pod status:

```
allns-x-amzerodowntimerollingdeployment-ri395v-0   alertmanager-webhook-78b9fd4cf5-4dvdf                   0/1     CrashLoopBackOff   3 (50s ago)   101s
```

* `alertmanager-webhook` pod logs: 

```
runtime: failed to create new OS thread (have 2 already; errno=22)
fatal error: newosproc

runtime stack:
runtime.throw(0x6884d4, 0x9)
	/usr/local/go/src/runtime/panic.go:596 +0x95
runtime.newosproc(0xc420026000, 0xc420036000)
	/usr/local/go/src/runtime/os_linux.go:163 +0x18c
runtime.newm(0x696d78, 0x0)
	/usr/local/go/src/runtime/proc.go:1628 +0x137
runtime.main.func1()
	/usr/local/go/src/runtime/proc.go:126 +0x36
runtime.systemstack(0x7b8c00)
	/usr/local/go/src/runtime/asm_amd64.s:327 +0x79
runtime.mstart()
	/usr/local/go/src/runtime/proc.go:1132

goroutine 1 [running]:
runtime.systemstack_switch()
	/usr/local/go/src/runtime/asm_amd64.s:281 fp=0xc420022788 sp=0xc420022780
runtime.main()
	/usr/local/go/src/runtime/proc.go:127 +0x6c fp=0xc4200227e0 sp=0xc420022788
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2197 +0x1 fp=0xc4200227e8 sp=0xc4200227e0
```

**Anything else we need to know?**:

--- a/.gitignore
+++ b/.gitignore
@@ -2,6 +2,7 @@
 /admission-webhook
 /po-lint
 /prometheus-config-reloader
+example/alertmanager-webhook/linux/
 .build/
 *~
 *.tgz

--- a/example/alertmanager-webhook/Dockerfile
+++ b/example/alertmanager-webhook/Dockerfile
@@ -1,0 +1,7 @@
+FROM scratch
+
+ARG ARCH="amd64"
+
+COPY linux/${ARCH}/alertmanager-webhook /bin/alertmanager-webhook
+
+ENTRYPOINT ["/bin/alertmanager-webhook"]

--- a/example/alertmanager-webhook/Makefile
+++ b/example/alertmanager-webhook/Makefile
@@ -1,0 +1,14 @@
+IMAGE ?= quay.io/prometheus-operator/prometheus-alertmanager-test-webhook
+TAG ?= latest
+
+.PHONY: build
+build:
+	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -tags netgo -o linux/amd64/ .
+	GOOS=linux GOARCH=arm64 CGO_ENABLED=0 go build -tags netgo -o linux/arm64/ .
+
+.PHONY: push
+push: build
+	buildah manifest create $(IMAGE):$(TAG)
+	buildah bud --arch amd64 --build-arg ARCH=amd64 --manifest $(IMAGE):$(TAG) -t $(IMAGE):$(TAG)-amd64 .
+	buildah bud --arch arm64 --build-arg ARCH=arm64 --manifest $(IMAGE):$(TAG) -t $(IMAGE):$(TAG)-arm64 .
+	buildah manifest push --all $(IMAGE):$(TAG) docker://$(IMAGE):$(TAG)

--- a/example/alertmanager-webhook/README.md
+++ b/example/alertmanager-webhook/README.md
@@ -1,0 +1,15 @@
+This directory contains a very simple program that can receive alert
+notifications from Alertmanager. It is used by the end-to-end tests to verify
+that Alertmanager works as expected.
+
+The program is available at `quay.io/prometheus-operator/prometheus-alertmanager-test-webhook:latest`.
+
+## Updating the image on quay.io
+
+The image requires very few updates since the program is very simple and only used for testing.
+
+Pre-requisites:
+* Credentials to push the image to `quay.io/prometheus-operator/prometheus-alertmanager-test-webhook`.
+* Buildah CLI + the `qemu-user-static` package.
+
+Running `make manifest` should be all that is needed.

--- a/example/alertmanager-webhook/main.go
+++ b/example/alertmanager-webhook/main.go
@@ -19,8 +19,13 @@ import (
 	"net/http"
 )
 
+const (
+	port = "5001"
+)
+
 func main() {
-	_ = http.ListenAndServe(":5001", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	fmt.Printf("Listening for Alertmanager notifications on :%s\n", port)
+	_ = http.ListenAndServe(":"+port, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		fmt.Println("Alertmanager Notification Payload Received")
 	}))
 }

--- a/example/alertmanger-webhook/Dockerfile
+++ b/example/alertmanger-webhook/Dockerfile
@@ -1,5 +1,0 @@
-FROM quay.io/prometheus/busybox:latest
-
-COPY main /bin/main
-
-ENTRYPOINT ["/bin/main"]

--- a/example/alertmanger-webhook/Makefile
+++ b/example/alertmanger-webhook/Makefile
@@ -1,3 +1,0 @@
-all:
-	CGO_ENABLED=0 go build --installsuffix cgo main.go
-	docker build -t quay.io/coreos/prometheus-alertmanager-test-webhook .

--- a/test/e2e/alertmanager_test.go
+++ b/test/e2e/alertmanager_test.go
@@ -528,7 +528,7 @@ func testAMZeroDowntimeRollingDeployment(t *testing.T) {
 					Containers: []v1.Container{
 						{
 							Name:  "webhook-server",
-							Image: "quay.io/coreos/prometheus-alertmanager-test-webhook",
+							Image: "quay.io/prometheus-operator/prometheus-alertmanager-test-webhook:latest",
 							Ports: []v1.ContainerPort{
 								{
 									Name:          "web",

